### DISCUSSION TITLE
2 CHILD_SA's and a decrypt error

### DISCUSSION THREAD

**Discussion by amodster** [strongswan/strongswan#1793] [General]
Hello

We are trying to figure out why there are 2 CHILD_SA's and a decrypt error.
I looked thru the strongswan support and found a few iteads on the 2 CHILD_SA, but not sure.

Setup:
	Windows IPSec server
	Strongswan IPSec client

2023 Jul 10 11:06:34+00:00 wglng-10766 charon [info] 12[ENC] DELETE verification failed
2023 Jul 10 11:06:34+00:00 wglng-10766 charon [info] 12[ENC] could not decrypt payloads
2023 Jul 10 11:06:34+00:00 wglng-10766 charon [info] 12[IKE] message verification failed
2023 Jul 10 11:06:34+00:00 wglng-10766 charon [info] 12[ENC] generating INFORMATIONAL response 1 [ N(INVAL_SYN) ]
[security.log.0.zip](https://github.com/strongswan/strongswan/files/12044073/security.log.0.zip)

**Comment by tobiasbrunner** [comment]
When verifying a delete payload, we check that the SPI size matches the protocol (4 for ESP, 0 for IKEv2, 16 for IKEv1) and that the SPI data has the correct length according to that and the number of SPIs (a delete payload for ESP can contain multiple). Since we don't know what exactly the peer sent here (you'd have to increase the log level for _enc_ to see that) it's difficult to say what it did incorrectly exactly.

It's also unclear why it created a new CHILD_SA before that. Or what the purpose of the delete was (but presumably it tried to delete the previous CHILD_SA). It wasn't a rekeying, at least not a proper one as that would contain a REKEY_SA notify. And the SA looks the same as the one created with IKE_AUTH, so it's not that rekeying was impossible because of some changed parameters. But maybe the other implementation can't do proper rekeyings and has to go the IKEv1 route and create new CHILD_SAs (which is technically what happens during a rekeying as well, but with IKEv2 and REKEY_SA notifies we know which SA is getting replaced so e.g. traffic interruptions can be largely avoided).
